pkg/ec/lint.go

```diff
@@ -68,33 +82,52 @@ func LintEmbeddedClusterVersion(specFiles domain.SpecFiles) ([]domain.LintExpres
 	return lintExpressions, nil
 }
 
-func checkIfECVersionExists(version string) (bool, error) {
-	url := "http://api.github.com/repos/replicatedhq/embedded-cluster/releases/tags/%s"
+func checkIfECVersionExists(version string) (*EmbeddedClusterVersion, bool, error) {
+	url := githubAPIURL + "/repos/replicatedhq/embedded-cluster/releases/tags/%s"
 	token := os.Getenv("GITHUB_API_TOKEN")
 	var bearer = "Bearer " + token
 
 	rwMutex.RLock()
-	verIsCached := ecVersions[version]
+	ecVersion, found := ecVersions[version]
 	rwMutex.RUnlock()
 
-	if !verIsCached {
-		req, err := http.NewRequest("GET", fmt.Sprintf(url, version), nil)
-		if err != nil {
-			return false, errors.Wrap(err, "failed to create new request")
-		}
-		req.Header.Set("Authorization", bearer)
-		client := &http.Client{}
-		resp, _ := client.Do(req)
-		if resp.StatusCode == 404 {
-			return false, nil
-		} else if resp.StatusCode == 200 {
-			rwMutex.Lock()
-			ecVersions[version] = true
-			rwMutex.Unlock()
-		} else {
-			return false, errors.New(fmt.Sprintf("received non 200 status code (%d) from GitHub API request", resp.StatusCode))
-		}
+	if found {
+		return &ecVersion, true, nil
+	}
+
+	req, err := http.NewRequest("GET", fmt.Sprintf(url, version), nil)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "failed to create new request")
 	}
+	req.Header.Set("Authorization", bearer)
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, false, errors.Wrap(err, "failed to make http request")
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode == http.StatusNotFound {
+		return nil, false, nil
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, false, errors.New(fmt.Sprintf("received non 200 status code (%d) from GitHub API request", resp.StatusCode))
+	}
+
+	var newVersion EmbeddedClusterVersion
+	if err := json.NewDecoder(resp.Body).Decode(&newVersion); err != nil {
+		return nil, false, errors.Wrap(err, "failed to decode embedded cluster version json")
+	}
+
+	if newVersion.PreRelease {
+		return &newVersion, true, nil
+	}
```

Comment:
this needs a comment - it looks like we're purposefully not caching prereleases, which I would agree with, but that should be explicitly listed

Reply:
Done.